src/Microsoft.Data.Sqlite/SqliteConnection.cs

```diff
@@ -238,10 +238,13 @@ protected override void Dispose(bool disposing)
         // NB: Other providers don't set Transaction
         public new virtual SqliteCommand CreateCommand() => new SqliteCommand { Connection = this, Transaction = Transaction };
         protected override DbCommand CreateDbCommand() => CreateCommand();
-        public new virtual SqliteTransaction BeginTransaction() => BeginTransaction(IsolationLevel.Unspecified);
-        protected override DbTransaction BeginDbTransaction(IsolationLevel isolationLevel) => BeginTransaction(isolationLevel);
+        public new virtual SqliteTransaction BeginTransaction() => BeginTransactionImpl(IsolationLevel.Unspecified);
+        protected override DbTransaction BeginDbTransaction(IsolationLevel isolationLevel) => BeginTransactionImpl(isolationLevel);
 
         public new virtual SqliteTransaction BeginTransaction(IsolationLevel isolationLevel)
+            => BeginTransactionImpl(isolationLevel);
+
+        internal SqliteTransaction BeginTransactionImpl(IsolationLevel isolationLevel, int commandTimeout = SqliteCommand.DefaultCommandTimeout)
```

Comment:
I'm against using internal methods for testing... Using the internal constructor of SqliteTransaction to ensure that command timeout is honored feels cleaner to me.